pgdog/src/backend/databases.rs

```diff
@@ -472,18 +454,59 @@ pub fn from_config(config: &ConfigAndUsers) -> Databases {
 
     let mut mirrors = HashMap::new();
 
-    for (source_user, source_cluster) in databases.iter() {
-        let mirror_clusters = databases
+    // Helper function to get users for a database
+    let get_database_users = |db_name: &str| -> std::collections::HashSet<&String> {
+        databases
             .iter()
-            .filter(|(mirror_user, mirror_cluster)| {
-                mirror_cluster.mirror_of() == Some(source_cluster.name())
-                    && mirror_user.user == source_user.user
-            })
-            .map(|(_, c)| c.clone())
-            .collect::<Vec<_>>();
-
-        if !mirror_clusters.is_empty() {
-            mirrors.insert(source_user.clone(), mirror_clusters);
+            .filter(|(_, cluster)| cluster.name() == db_name)
+            .map(|(user, _)| &user.user)
+            .collect()
+    };
+
+    // Validate mirroring configurations and collect valid ones
+    let mut valid_mirrors = std::collections::HashSet::new();
+
+    for mirror_config in &config.config.mirroring {
+        let source_users = get_database_users(&mirror_config.source_db);
+        let dest_users = get_database_users(&mirror_config.destination_db);
+
+        if !source_users.is_empty() && !dest_users.is_empty() && source_users == dest_users {
```

Comment:
We shouldn't insert mirrors with no users: their pools wont be activated.

Reply:
I believe this checks that both source_users and dest_users are not empty, unless I'm misunderstanding what you're saying?

Reply:
You're right, I think it was late when I read this :)